suse_migration_services/migration_config.py

```diff
@@ -0,0 +1,63 @@
+# Copyright (c) 2019 SUSE Linux LLC.  All rights reserved.
+#
+# This file is part of suse-migration-services.
+#
+# suse-migration-services is free software: you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# suse-migration-services is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with suse-migration-services. If not, see <http://www.gnu.org/licenses/>
+#
+import yaml
+
+# project
+from suse_migration_services.defaults import Defaults
+
+
+class MigrationConfig(object):
+    """
+    **Implements reading of migration configuration file:**
+
+    /etc/migration-config.yml
+
+    The migration config file lives inside of the migration
+    live image system. It is usually provided as a pre-setup
+    version as part of the live migration image build.
+    """
+    target_products = {
+        'ses': 'SLES/15.1/x86_64',
+        'sles': 'SLES/15/x86_64'
+    }
+
+    def __init__(self):
+        self.config_data = None
+        with open(Defaults.get_migration_config_file(), 'r') as config:
+            self.config_data = yaml.safe_load(config)
+
+    def get_migration_product(self):
+        """
+        Returns the full qualified product
+
+        The value returned is passed to the --product option in the
+        zypper migration call
+
+        migration:
+          target:
+            product: product_name
+        """
+        migration = self.config_data.get('migration')
```

Comment:
Let's call the config option "migration_product". I don't think we need sub-options for the migration as is currently implemented. Based on teh current implementation it would look as follows:

```
migration:
  - target: sles
```

I think

```
migration_product: sles
```

is sufficient. Then the code simplifies to:
```
migration_product = self.config_data.get('migration_product', 'sles')
```

Reply:
Fine by me

how does the red cape fit on :))

Reply:
I have removed the sub-options, when/if approved, I will update ```/etc/migration-config.yml``` file we provide to match this structure.